tests/comfy_cli/cloud/test_login_command.py

```
@@ -57,6 +57,52 @@ def _no_tracking(monkeypatch):
+    monkeypatch.setattr(command, "run_login", fake_run_login)
+    monkeypatch.setattr(command.store, "save_cloud_session", save_then_record)
+    monkeypatch.setattr(command.knowledge, "refresh_if_stale", lambda: order.append("warm"))
+    set_renderer(Renderer(mode=OutputMode.JSON, command="cloud login", version="test"))
+
+    command.login_cmd(no_browser=True, timeout=300)
+
+    assert order == ["save", "warm"]
+    assert _parse_lines(capsys.readouterr().out)[-1]["ok"] is True
+
+
+def test_login_timeout_does_not_warm_the_knowledge_cache(monkeypatch, capsys, _knowledge_warm):
```

Pylint (4.0.7)

[convention] Missing function or method docstring

(C0116)